### PR TITLE
[docs] fix build

### PR DESCRIPTION
**Pull request by maxpumperla**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/api/dataset.rst [hunk 1]
```diff
@@ -149,3 +149,17 @@ Serialization
    Dataset.has_serializable_lineage
    Dataset.serialize_lineage
    Dataset.deserialize_lineage
+
+
```
Comment (maxpumperla) [inline doc/source/data/api/dataset.rst:153]: We need these to be explicitly in a TOC somewhere. We can argue about the exact position after fixing the build.

**Review by angelinalg — APPROVED** [review]
LGTM

File: doc/source/rllib/package_ref/rl_modules.rst [hunk 2]
```diff
@@ -114,7 +114,6 @@ Constructor
     :toctree: doc/
 
     MultiAgentRLModule
```
Comment (kouroshHakha) [inline doc/source/rllib/package_ref/rl_modules.rst:116]:
```suggestion
    MultiAgentRLModule
    MultiAgentRLModule.setup()
```

**Review by kouroshHakha — APPROVED** [review]
Thanks @maxpumperla. The build method got axed and got replaced with the setup method due to conflict with internal tf.keras.Model API. If you can replace that it will be great. Thanks.

File: doc/source/data/api/dataset.rst [hunk 3]
```diff
@@ -149,3 +149,17 @@ Serialization
...
+Internals
+---------
+
+.. autosummary::
+    :toctree: doc/
```
Comment (ericl) [inline doc/source/data/api/dataset.rst:158]: Oh cool, this deprecates most of my fix PR here: https://github.com/ray-project/ray/pull/34228

Btw, is there a way to hide this internals section by default? Some of these are legacy backwards compatibility aliases that we don't want to expose.

File: python/ray/data/_internal/execution/interfaces.py [hunk 4]
```diff
@@ -191,22 +191,6 @@ class ExecutionOptions:
     """Common options for execution.
 
     Some options may not be supported on all executors (e.g., resource limits).
-
```
Comment (ericl) [inline python/ray/data/_internal/execution/interfaces.py:194]: I was able to fix this via adding the nosummary template: https://github.com/ray-project/ray/pull/34228/files#diff-b3a1d55836e0b2035d287e569a716fa02296221b15eee9babd6c2ea0f62a2ec7R13

File: doc/source/data/api/dataset.rst (hunk 3 above)
Comment (maxpumperla) [reply doc/source/data/api/dataset.rst:158]: @ericl yes, toctrees can be `:hidden:`, if this is what we want.

File: python/ray/data/_internal/execution/interfaces.py (hunk 4 above)
Comment (maxpumperla) [reply python/ray/data/_internal/execution/interfaces.py:194]: cool, thanks

**Review by scv119 — APPROVED** [review]

**Comment by pcmoritz** [comment]
^ I tried to fix the merge conflicts -- let's see if it passes and if yes we can merge it :)

cc @maxpumperla

File: python/ray/data/_internal/execution/interfaces.py (hunk 4 above)
Comment (ericl) [reply python/ray/data/_internal/execution/interfaces.py:194]: @pcmoritz you can undo this particular diff now (fixed in master).

Comment (pcmoritz) [reply python/ray/data/_internal/execution/interfaces.py:194]: Thanks, will do!

**Comment by pcmoritz** [comment]
Merging this now -- there is one more documentation failure that will be fixed by the RL team. Thanks for fixing all this @maxpumperla :)

**Comment by maxpumperla** [comment]
@pcmoritz thank you for jumping in and resolving the conflicts!

**Pull request merged**